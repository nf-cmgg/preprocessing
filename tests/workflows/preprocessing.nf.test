nextflow_workflow {

    name "Test Workflow PREPROCESSING"
    script "workflows/preprocessing.nf"
    workflow "PREPROCESSING"

    tag "workflows"
    tag "workflows/preprocessing"

    test("preprocessing - fastq - bwa - bamsormadup - roi") {

        when {
            workflow {
                """
                // ch_samplesheet
                input[0] = Channel.of(
                    [
                        [                           // meta
                            id: "sample1",
                            samplename: "sample1",
                            library: "test",
                            organism: "Homo sapiens",
                            tag: "WES",
                            sample_type: "DNA"
                        ],
                        //fastq_1
                        file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/fastq/sample1_R1.fastq.gz", checkIfExists: true),
                        // fastq_2
                        file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/fastq/sample1_R2.fastq.gz", checkIfExists: true),
                        [],                         // samplesheet
                        [],                         // sampleinfo
                        [],                         // flowcell

                    ]
                )
                // genomes
                input[1] = [
                    GRCh38: [
                        fasta: "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna",
                        fai: "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna.fai",
                        dict: "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.dict",
                        bwamem: "s3://test-data/genomics/homo_sapiens/genome/bwa/",
                        gtf: "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.gtf"
                    ]
                ]
                // aligner
                input[2] = "bwamem"
                // markdup
                input[3] = "bamsormadup"
                // roi
                input[4] = "https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/regions/roi_chr21.bed"
                input[5] = null
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                file(workflow.out.multiqc_report[0][0]).name,
                workflow.out.versions
            ).match()
        }
    }

    test("preprocessing - fastq - bwa - bamsormadup - no roi") {

        when {
            workflow {
                """
                // ch_samplesheet
                input[0] = Channel.of(
                    [
                        [                           // meta
                            id: "sample1",
                            samplename: "sample1",
                            library: "test",
                            organism: "Homo sapiens",
                            tag: "WGS",
                            sample_type: "DNA"
                        ],
                        //fastq_1
                        file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/fastq/sample1_R1.fastq.gz", checkIfExists: true),
                        // fastq_2
                        file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/fastq/sample1_R2.fastq.gz", checkIfExists: true),
                        [],                         // samplesheet
                        [],                         // sampleinfo
                        [],                         // flowcell

                    ]
                )
                // genomes
                input[1] = [
                    GRCh38: [
                        fasta: "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna",
                        fai: "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna.fai",
                        dict: "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.dict",
                        bwamem: "s3://test-data/genomics/homo_sapiens/genome/bwa/",
                        gtf: "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.gtf"
                    ]
                ]
                // aligner
                input[2] = "bwamem"
                // markdup
                input[3] = "bamsormadup"
                // roi
                input[4] = ""
                input[5] = null
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                file(workflow.out.multiqc_report[0][0]).name,
                workflow.out.versions
            ).match()
        }
    }

    test("preprocessing - fastq - bwa - bamsormadup - roi - no coverage/no picard") {
        when {
            params {
                run_coverage = false
                disable_picard_metrics = true
            }
            workflow {
                """
                // ch_samplesheet
                input[0] = Channel.of(
                    [
                        [                           // meta
                            id: "sample1",
                            samplename: "sample1",
                            library: "test",
                            organism: "Homo sapiens",
                            tag: "WES",
                            sample_type: "DNA"
                        ],
                        //fastq_1
                        file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/fastq/sample1_R1.fastq.gz", checkIfExists: true),
                        // fastq_2
                        file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/fastq/sample1_R2.fastq.gz", checkIfExists: true),
                        [],                         // samplesheet
                        [],                         // sampleinfo
                        [],                         // flowcell

                    ]
                )
                // genomes
                input[1] = [
                    GRCh38: [
                        fasta: "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna",
                        fai: "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna.fai",
                        dict: "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.dict",
                        bwamem: "s3://test-data/genomics/homo_sapiens/genome/bwa/",
                        gtf: "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.gtf"
                    ]
                ]
                // aligner
                input[2] = "bwamem"
                // markdup
                input[3] = "bamsormadup"
                // roi
                input[4] = "https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/regions/roi_chr21.bed"
                input[5] = null
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                file(workflow.out.multiqc_report[0][0]).name,
                workflow.out.versions
            ).match()
        }
    }
}
