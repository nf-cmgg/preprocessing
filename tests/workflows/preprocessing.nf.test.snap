{
    "preprocessing - fastq - bwa - bamsormadup - roi": {
        "content": [
            "multiqc_report.html",
            [
                "versions.yml:md5,13e87ed415a2e15f5ee6cd61a696f159",
                "versions.yml:md5,2789454a6d07582f30cca7623d431a2b",
                "versions.yml:md5,321e55c8f19102dc87a10e981635edda",
                "versions.yml:md5,4257181c63c732a41de1cb83471b289b",
                "versions.yml:md5,4cadd59c9520ebd940030a4cc6d39d64",
                "versions.yml:md5,71f9d3ee1ac829a35187963f25c7c582",
                "versions.yml:md5,a3854dfc77096728996132543bfa9c39",
                "versions.yml:md5,d11c133ecb39ba9f6d7e081a8a6ff868",
                "versions.yml:md5,f16d4e5179f6538e258272e7a8fa3a2c",
                "versions.yml:md5,fcbb82b85126720b43a2f8380bcf2f81"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-15T13:02:45.455557"
    },
    "preprocessing - fastq - bwa - bamsormadup - roi - no coverage/no picard": {
        "content": [
            "multiqc_report.html",
            [
                "versions.yml:md5,13e87ed415a2e15f5ee6cd61a696f159",
                "versions.yml:md5,2789454a6d07582f30cca7623d431a2b",
                "versions.yml:md5,321e55c8f19102dc87a10e981635edda",
                "versions.yml:md5,4257181c63c732a41de1cb83471b289b",
                "versions.yml:md5,71f9d3ee1ac829a35187963f25c7c582",
                "versions.yml:md5,d11c133ecb39ba9f6d7e081a8a6ff868",
                "versions.yml:md5,f16d4e5179f6538e258272e7a8fa3a2c",
                "versions.yml:md5,fcbb82b85126720b43a2f8380bcf2f81"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-15T13:13:24.558576"
    },
    "preprocessing - fastq - bwa - bamsormadup - no roi": {
        "content": [
            "multiqc_report.html",
            [
                "versions.yml:md5,13e87ed415a2e15f5ee6cd61a696f159",
                "versions.yml:md5,2789454a6d07582f30cca7623d431a2b",
                "versions.yml:md5,321e55c8f19102dc87a10e981635edda",
                "versions.yml:md5,4257181c63c732a41de1cb83471b289b",
                "versions.yml:md5,4cadd59c9520ebd940030a4cc6d39d64",
                "versions.yml:md5,71f9d3ee1ac829a35187963f25c7c582",
                "versions.yml:md5,d11c133ecb39ba9f6d7e081a8a6ff868",
                "versions.yml:md5,f16d4e5179f6538e258272e7a8fa3a2c",
                "versions.yml:md5,f86ff6a955bce1589780c33ee28a1126",
                "versions.yml:md5,fcbb82b85126720b43a2f8380bcf2f81"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-15T13:12:00.645316"
    }
}